src/Microsoft.TemplateEngine.Cli/PostActionProcessors/ProcessStartPostActionProcessor.cs

```diff
@@ -60,9 +68,37 @@ public bool Process(IEngineEnvironmentSettings environment, IPostAction actionCo
 
                 if (commandResult.ExitCode != 0)
                 {
-                    string error = commandResult.StandardError.ReadToEnd();
                     Reporter.Error.WriteLine(LocalizableStrings.CommandFailed);
-                    Reporter.Error.WriteLine(string.Format(LocalizableStrings.CommandOutput, error));
+                    if (redirectStandardOutput || redirectStandardError)
+                    {
+                        Reporter.Error.WriteLine(LocalizableStrings.CommandOutput);
+                    }
+                    if (redirectStandardOutput)
```

Comment:
minor: consider consolidating logic for both std output & std error.
`            void X(string stdType, bool item)
`            {
`                if (item)
`                {
`                    string output = commandResult.StandardOutput.ReadToEnd();
`                    Reporter.Error.WriteLine($"{stdType}:");
`                    if (string.IsNullOrWhiteSpace(output))
`                    {
`                        Reporter.Error.WriteLine(LocalizableStrings.Generic_Empty);
`                    }
`                    else
`                    {
`                        Reporter.Error.WriteLine(output);
`                    }
`                }
`            }

Reply:
I switched to extensions in last commit, I prefer that way more for this purpose. Problem with function above that it needs also to know what property to take from result, and it gets messier for such a small chunk of code.